backend/src/main/java/de/neuefische/repository/TodoRepo.java

```diff
@@ -0,0 +1,40 @@
+package de.neuefische.repository;
+
+import de.neuefische.TodoElement;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class TodoRepo {
+    private final List<TodoElement> todoList = new ArrayList<>();
+
+    public List<TodoElement> getTodoList(){
+        return todoList;
+    }
+
+    public Optional<TodoElement> getTodoElementByTitle(String title){
+        return todoList.stream().filter(e -> e.getTitle().equals(title)).findFirst();
+    }
+
+    public List<TodoElement> getAllCompletedTodoElements(){
+        return getTodoList().stream().filter(TodoElement::isState).toList();
+    }
+
+    public List<TodoElement> getAllUncompletedTodoElements(){
+        return getTodoList().stream().filter(e -> !e.isState()).toList();
+    }
+
+    public void addTodo(TodoElement todoElement) {
+        todoList.add(todoElement);
+    }
+
+    public void  setStateCompleted(String id) {
+        todoList.stream()
```

Comment:
Das Repo ist verantwortlich fürs Ablegen und Ermitteln von `TodoElement`s. Das ein `TodoElement` den Zustand ändern kann ist für mich Business-Logik, die in den Service gehört.

Reply:
done